server/medical-contact/middleware.ts

```diff
@@ -0,0 +1,115 @@
+import type {Request, Response, NextFunction} from 'express';
+import {Types} from 'mongoose';
+import MedicalContactCollection from './collection';
+
+/**
+ * Checks if a medical contact with medicalContactId in req.params exists
+ */
+const isMedicalContactExists = async (req: Request, res: Response, next: NextFunction) => {
+    const validFormat = Types.ObjectId.isValid(req.params.medicalContactId);
+    const medicalContact = validFormat ? await MedicalContactCollection.findOne(req.params.medicalContactId) : '';
+    if (!medicalContact) {
+      res.status(404).json({
+        error: `Medical contact with medical contact ID ${req.params.medicalContactId} does not exist.`
+      });
+      return;
+    }
+
+  next();
+  
+};
+
+/**
+ * Checks if a medical contact with medicalContactId in req.params is active
+ */
+ const isMedicalContactActive = async (req: Request, res: Response, next: NextFunction) => {
+  const validFormat = Types.ObjectId.isValid(req.params.medicalContactId);
+  const medicalContact = validFormat ? await MedicalContactCollection.findOne(req.params.medicalContactId) : '';
+  if (!medicalContact.active) {
+    res.status(409).json({
+      error: `Medical contact with medical contact ID ${req.params.medicalContactId} is already deactivated.`
+    });
+    return;
+  }
+
+next();
+
+};
+
+/**
+ * Checks if a first name in req.body is valid, that is, it matches the name regex
+ */
+const isValidFirstName = (req: Request, res: Response, next: NextFunction) => {
+    const nameRegex = /^\w+$/i;
+    if (req.body.first_name){
+      if (!nameRegex.test(req.body.first_name)) {
+        res.status(400).json({
+        error: 'First name must be a nonempty alphanumeric string.'
+        });
+        return;
+    }
+    }
+
+    next();
+};
+
+/**
+ * Checks if a last name in req.body is valid, that is, it matches the name regex
+ */
+ const isValidLastName = (req: Request, res: Response, next: NextFunction) => {
+    const nameRegex = /^\w+$/i;
+    if (req.body.last_name){
+      if (!nameRegex.test(req.body.last_name)) {
+        res.status(400).json({
+          error: 'Last name must be a nonempty alphanumeric string.'
+        });
+        return;
+      }
+    }
+  
+    next();
+};
+
+/**
+ * Checks if a phone number in req.body is valid, that is, it matches the phone number regex
+ */
+ const isValidPhoneNumber = (req: Request, res: Response, next: NextFunction) => {
+    const phoneRegex = /^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]\d{3}[\s.-]\d{4}$/;
```

Comment:
phone number formatting is a notoriously annoying thing

I usually just accept anything, but I think this regex is flexible enough to be sufficient.